apollo-federation/src/sources/connect/json_selection/README.md

```diff
@@ -81,15 +81,17 @@ worry if it doesn't seem helpful yet, as every rule will be explained in detail
 below.
 
 ```ebnf
-JSONSelection        ::= NakedSubSelection | PathSelection
+JSONSelection        ::= PathSelection | NakedSubSelection
 NakedSubSelection    ::= NamedSelection* StarSelection?
 SubSelection         ::= "{" NakedSubSelection "}"
-NamedSelection       ::= NamedPathSelection | NamedFieldSelection | NamedGroupSelection
+NamedSelection       ::= NamedPathSelection | PathWithSubSelection | NamedFieldSelection | NamedGroupSelection
 NamedPathSelection   ::= Alias PathSelection
 NamedFieldSelection  ::= Alias? Key SubSelection?
 NamedGroupSelection  ::= Alias SubSelection
 Alias                ::= Key ":"
-PathSelection        ::= (VarPath | KeyPath | AtPath | ExprPath) SubSelection?
+Path                 ::= VarPath | KeyPath | AtPath | ExprPath
+PathSelection        ::= Path SubSelection?
+PathWithSubSelection ::= Path SubSelection
 VarPath              ::= "$" (NO_SPACE Identifier)? PathStep*
 KeyPath              ::= Key PathStep+
 AtPath               ::= "@" PathStep*
```

Comment:
A good place to start reviewing this PR is the `README.md`, especially these changes to the formal grammar.